packages/components/src/custom-select/index.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+/**
+ * Internal dependencies
+ */
+// import { type CustomSelectProps } from './types';
+import { CustomSelectButton, CustomSelectPopover } from './styles';
+import CustomSelectItem from './custom-select-item';
+import type { CustomSelectProps } from './types';
+
+function CustomSelect( props: CustomSelectProps ) {
+	const { children, defaultValue, label, onChange } = props;
+
+	const store = Ariakit.useSelectStore( {
+		setValue: ( nextValue ) => onChange?.( nextValue ),
+		defaultValue,
+	} );
+
+	const { value } = store.useState();
+
+	return (
+		<>
+			<Ariakit.SelectLabel store={ store }>{ label }</Ariakit.SelectLabel>
+			<CustomSelectButton store={ store }>
+				{ typeof defaultValue === 'object' ? defaultValue : value }
```

Comment:
I'm sure there is a better solution, but this seems to fit for now. And it's likely to change when exploring controlled vs uncontrolled. 

If the `defaultValue` isn't a `string` or `undefined`, then it needs to be added a child to the Select button (`Ariakit.Select` styled as `CustomSelectButton`). If it is a `string` or left `undefined`, then it cannot be added as a child unless the state is controlled. Otherwise, it would be overwritten by the default value. 

I have it comparing against an `object` as I believe there may be other types I haven't accounted for. This will likely make for a good test to add.

Reply:
Resolved in https://github.com/WordPress/gutenberg/pull/55790/commits/56d9f7fb418e2a1c07032a4164cc6790784b7337 with @ciampo